PostAppTest/SocialPostTest.cs

```diff
@@ -0,0 +1,148 @@
+using Moq;
+using Server.Controllers;
+using Server.Models;
+using Server.Repository;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace SocialPostsTest
+{
+    public class SocialPostControllerTest
+    {
+
+        [Fact]
+        public void GetAllSocialPost_Returns_All_posts()
+        {
+            // Arrange
+            var mockRepo = new Mock<IPostRepository>();
+
+            mockRepo.Setup(repo => repo.GetSocialPost())
+            .Returns(GetTestPost());
+
+            var controller = new SocialPostController(mockRepo.Object);
+
+            var result = controller.GetSocialPost();
+
+            Assert.NotNull(result);
+        }
+
+        [Fact]
+        public void GetSocialPostById_Returns_Post()
+        {
+            var id = 1;
+
+           
+            var mockRepo = new Mock<IPostRepository>();
+
+            mockRepo.Setup(repo => repo.GetSocialPostById(id))
+            .Returns(GetTestPostById());
+
+            var controller = new SocialPostController(mockRepo.Object);
+
+            var result = controller.GetSocialPostById(1);
+
+            Assert.NotNull(result);
+        }
+
+        [Fact]
+        public void UpdateSocialPost_Returns_Post()
+        {
+            var id = 1;
+
+           
+            var mockRepo = new Mock<IPostRepository>();
+
+            mockRepo.Setup(repo => repo.UpdateSocialPost(id, GetTestPostById()))
+            .Returns(GetTestPostById());
+
+            var controller = new SocialPostController(mockRepo.Object);
+
+            var result = controller.UpdateSocialPost(id, GetTestPostById());
+
+            Assert.NotNull(result);
+        }
+
+        [Fact]
+        public void UpdateSocialPostLike_returns_int()
+        {
+            var id = 1;
```

Comment:
Create fixture file and use common constants and functions from there for all test cases

Reply:
created fixture file